src/network/httpRequest.js

```diff
@@ -0,0 +1,42 @@
+import fetch from 'node-fetch';
+
+/**
+ * Executes an HTTP request.
+ * @param url {String} The URL we want to request.
+ * @param options {Object} The fetch options.
+ * @return {Object} The response data as a Promise.
+ */
+export function request(url, options) {
+  console.log('url:', url, 'options:', options);
+
+  return fetch(url, options)
+    .then(checkStatusCode)
+    .then(parseResponse);
+}
+
+function checkStatusCode(response) {
+  if (response.status >= 200 && response.status < 300) {
+    return Promise.resolve(response);
+  }
+  return Promise.reject(new Error(response.statusText));
+}
+
+/**
+ * Parses the HTTP response returned by a network request.
+ * @param response {Object} A response from a network request.
+ * @return {Object} The parsed JSON from the request.
+ */
+function parseResponse(response) {
+  if (response.url === 'https://testnet.qtum.org/insight-api/statistics/total') {
```

Comment:
why do we need this? it returns a json

Reply:
The response is it GZIP format so I need uncompress and read it with `response.text()` first, then parse with `JSON.parse()`.

Let me make this more generic and detect if it is a GZIP instead of using the url.

Reply:
Nevermind. This was user error (nothing to do with GZIP). I was trying to return `res.result` when that didn't exist for the insight-api. Fixed to return the correct payload.